src/utils/common.js

```diff
@@ -591,18 +591,26 @@ export const getDeviceWidth = () => {
   return Dimensions.get('window').width;
 };
 
-export const formatTransactionFee = (feeInWei: string | number, gasToken: ?GasToken): string => {
-  if (!feeInWei) return '';
+export const getFormattedTransactionFeeValue = (feeInWei: string | number | BigNumber, gasToken: ?GasToken): string => {
+  // fixes exponential values with BigNumber.toPrecision()
+  const parsedFeeInWei = typeof feeInWei === 'object' // BigNumber
+    ? feeInWei.toPrecision()
+    : feeInWei.toString();
 
   if (gasToken && !isEmpty(gasToken)) {
-    const { symbol, decimals } = gasToken;
-    return t('tokenValue', {
-      value: formatAmount(utils.formatUnits(feeInWei.toString(), decimals), 2),
-      token: symbol,
-    });
+    return formatAmount(utils.formatUnits(parsedFeeInWei, gasToken.decimals), 2);
```

Comment:
So here we're returning the rounded value, but later we use the real value for validation.. could that cause a problem when you have a bit less than a fee displayed?

Reply:
Not related to this PR, but we can definitely change the rounding to default, however, I'm not sure I remember why we adde rounding here in first place.

Reply:
Yep, that could be checked after that PR gets merged

Reply:
👍